kiali.go

```diff
@@ -155,9 +155,10 @@ func waitForSecret() {
 		}
 	}()
 	secret := <-foundSecretChan
+
 	cfg := config.Get()
-	cfg.Server.Credentials.Username = secret.Username
-	cfg.Server.Credentials.Passphrase = secret.Passphrase
+	cfg.Server.Credentials = security.CredentialList{security.Credentials{Username: secret.Username, Passphrase: secret.Passphrase}}
```

Comment:
I don't think this is what you want.

This code here sleeps until a secret is detected (this allows Kiali to start up first, and the user can create the secret later). Before, the secret only had a single username/passphrase in it - so when the secret was mounted, the Credentials (single user/pass) was sent to the channel. The code outside the loop then reads the single credentials found in the channel and stores it in the cfg.Server.Credentials as a kind of cache - so the authentication code doesn't have to keep cracking open the secret everytime we need to authenticate a user request.

But this new code appears to be creating a CredentialsList with just that single item still (secret.Username, secret.Passphrase). But now that we support multiple ones, shouldn't the cfg.Server.Credentials be the full list of all the names found in the secret? This code seems to only support just a single set of credentials?

Reply:
That's true, I changed this to get something compiling but the tests/compilation did not catch it.

Reply:
Maybe not for this PR, but when the full effort you are working on is done, update the molecule test found here to test this (this molecule tests that the read-only and read/write roles are created properly - we can enhance this test to actually make sure we can log in with these different users):

https://github.com/kiali/kiali/blob/master/operator/molecule/roles-test

This test here is for the LDAP stuff - but it provides an example how to log in as different users within the molecule framework:

https://github.com/kiali/kiali/tree/master/operator/molecule/ldap-test